File: VB/XtraReport1.vb
Imports System
Imports System.Drawing
Imports DevExpress.XtraReports.UI

Namespace BeforePrintEvent

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub xrLabel_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs)
            ' Obtain the current label.
            Dim label As XRLabel = CType(sender, XRLabel)
            ' Get the total value.
            Dim total As Double = Convert.ToDouble(GetCurrentColumnValue("Total"))
            ' Customize the label's appearance.
            If total < 100 Then
                label.ForeColor = Color.White
                label.BackColor = Color.Red
            ElseIf total > 1000 Then
                label.ForeColor = Color.White
                label.BackColor = Color.Green
            Else
                label.ForeColor = Color.Black
                label.BackColor = Color.Transparent
            End If
        End Sub
    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using DevExpress.XtraReports.UI;

namespace BeforePrintEvent {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void xrLabel_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            // Obtain the current label.
            XRLabel label = (XRLabel)sender;
            
            // Get the total value.
            double total = Convert.ToDouble(GetCurrentColumnValue("Total"));
            
            // Customize the label's appearance.
            if (total < 100) {
                label.ForeColor = Color.White;
                label.BackColor = Color.Red;
            }
            else if (total > 1000) {
                label.ForeColor = Color.White;
                label.BackColor = Color.Green;
            }
            else {
                label.ForeColor = Color.Black;
                label.BackColor = Color.Transparent;
            }
        }

    }
}
